homeassistant/components/climate/systemair_savecair.py

```diff
@@ -0,0 +1,269 @@
+
+    @property
+    def current_temperature(self):
+        """Return the current temperature."""
+        if "pdm_input_temp_value" not in self._client.data:
+            return None
+
+        current_temperature = self._client.data["pdm_input_temp_value"]
+
+        if current_temperature is None:
+            return None
+
+        return current_temperature / 10
+
+    @property
+    def target_temperature(self):
+        """Return the temperature we try to reach."""
+        if "main_temperature_offset" not in self._client.data:
+            return None
+
+        target_temperature = self._client.data["main_temperature_offset"]
+
+        if target_temperature is None:
+            return None
+
+        return target_temperature / 10
+
+    @property
+    def current_humidity(self):
+        """Return the current humidity."""
+        if "rh_sensor" not in self._client.data:
+            return None
+
+        return self._client.data["rh_sensor"]
+
+    @property
+    def current_fan_mode(self):
+        """Return the fan setting."""
+        if "main_airflow" not in self._client.data:
+            return None
+
+        if self._client.data["main_airflow"] is None:
+            return None
+
+        fan_mode = int(self._client.data["main_airflow"])
+        ha_fan_mode = SAVECAIR_FAN_MODES[fan_mode]
+
+        return ha_fan_mode
+
+    @property
+    def current_operation(self):
+        """Return current operation ie. heat, cool, idle."""
+        return self._client.get_current_operation()
+
+    @property
+    def is_on(self):
+        """Return true if the device is on."""
+        if "main_airflow" not in self._client.data:
+            return True
+
+        if self._client.data["main_airflow"] is None:
+            return False
+        airflow = self._client.data["main_airflow"]
+
+        if airflow is not None:
+            return int(self._client.data["main_airflow"]) - 1 > 0
+
+        return False
+
+    @property
+    def operation_list(self):
+        """Return the list of available operation modes."""
+        return self._operation_list
+
+    @property
+    def fan_list(self):
+        """Return the list of available fan modes."""
+        if self.current_operation != STATE_MANUAL:
+            return None
+
+        return self._fan_list
+
+    @property
+    def target_temperature_step(self):
+        """Return the supported step of target temperature."""
+        return 1
+
+    async def async_set_temperature(self, **kwargs):
+        """Set new target temperatures."""
+        temperature = kwargs.get(ATTR_TEMPERATURE)
+
+        await self._client.set_temperature(int(temperature))
+
+    async def async_set_fan_mode(self, fan_mode):
+        """Set new target temperature."""
+        await self._client.set_fan_mode(fan_mode)
+
+    async def async_set_operation_mode(self, operation_mode):
+        """Set the operation mode."""
+        await self._client.set_operation_mode(operation_mode)
```

Comment:
Operation mode is one of the home assistant climate base component states. Use a dict to convert here too.